flw/src/main/java/org/motechproject/nms/flw/service/impl/FrontLineWorkerServiceImpl.java

```diff
@@ -103,8 +194,9 @@ public void deleteAllowed(FrontLineWorker frontLineWorker) {
             throw new IllegalStateException("FLW in invalid state with null invalidation date");
         }
 
-        if (Math.abs(Weeks.weeksBetween(now, frontLineWorker.getInvalidationDate()).getWeeks()) < 6) {
-            throw new IllegalStateException("FLW must be in invalid state for 6 weeks before deleting");
+        if (Math.abs(Weeks.weeksBetween(now, frontLineWorker.getInvalidationDate()).getWeeks()) < weeksToKeepInvalidFLWs) {
+            throw new IllegalStateException(String.format("FLW must be in %s state for %s weeks before deleting",
```

Comment:
I originally did that but then decided I wanted to log exactly the same as what I was comparing against.  I should change line 194